### DISCUSSION TITLE
Does anybody know a linter that reports deprecated modules.

### DISCUSSION THREAD

**Discussion by zbindenren** [golangci/golangci-lint#2598] [Q&A] [answered]
In my repo I have:

```console
$ go get .
go: module git.pnet.ch/zbindenren/depricon is deprecated: The latest supported version is github.com/peterbourgon/ff/v2.

$ echo $status
0
```

```console
$ golangci-lint run --enable-all
WARN [runner] golangci-lint run --enable-all
WARN [runner] The linter 'interfacer' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.  
WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive. 
WARN [runner] The linter 'maligned' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.  Replaced by govet 'fieldalignment'. 
WARN [runner] The linter 'scopelint' is deprecated (since v1.39.0) due to: The repository of the linter has been deprecated by the owner.  Replaced by exportloopref. 

$ golangci-lint --version
golangci-lint has version 1.44.0 built from 617470fa on 2022-01-25T11:31:17Z
```

Would be great if `golangci-lint` reported this like other deprecation warnings.

Hase someone a solution for this?

**Comment by Antonboom** [comment, marked as answer]
@zbindenren, hi!

I think such a linter will be different from existing in `golangci-lint` (e.g. [depguard](https://github.com/OpenPeeDeeP/depguard) or [gomodguard](https://github.com/ryancurrah/gomodguard)). Because it will not work with AST, but it will need to find the dependency in $GOPATH and parse it `go.mod` (to find `// Deprecated: ...`, based on [spec](https://go.dev/ref/mod#go-mod-file-module)).

I'm not sure there aren't any pitfalls here.

Isn't it easier to add a check to CI based on the output from `go get` or `go list`?
Example: https://stackoverflow.com/a/71217351
